### PR TITLE
Fix an issue where cookies with an empty string were not stored.

### PR DESCRIPTION
**Pull request by peckrob**
This is a re-creation of PR #1991, with the requested fix in it and based off master.

Cookies with empty string values are technically a violation of RFC 6265, but this issue is known to occur in the wild (I personally encountered it in dealing with a SAP application server). Browsers handle it property.

Test was also fixed an an additional test for empty string was added.

### CODE REVIEW

File: src/Cookie/SetCookie.php [hunk 1]
```diff
@@ -392,9 +392,10 @@ public function validate()
                 . 'following characters: ()<>@,;:\"/?={}';
         }
 
-        // Value must not be empty, but can be 0
+        // Value must not be null. 0 and empty string are valid. Empty strings
```
Comment (GrahamCampbell) [inline src/Cookie/SetCookie.php:395]: These comments about zero are kinda stupid (not your fault). They must have been added after someone used empty by mistake instead of comparing against the empty string...

Comment (Nyholm) [reply src/Cookie/SetCookie.php:395]: I agree. But Im fine with keeping them.
